theano/sandbox/cuda/opt_util.py

```diff
@@ -126,3 +126,48 @@ def opt(node):
                 return maker(targ, *inputs)
         return opt
     return wrapper
+
+
+def pad_dims(input, leftdims, rightdims):
+    """Reshapes the input to a (leftdims + rightdims) tensor
+
+    This helper function is used to convert pooling inputs with arbitrary
+    non-pooling dimensions to the correct number of dimensions for the
+    GPU pooling ops.
+
+    This reduces or expands the number of dimensions of the input to
+    exactly `leftdims`, by adding extra dimensions on the left or by
+    combining some existing dimensions on the left of the input.
```

Comment:
I don't understand what this does from this descriptions. I know what it does because I know the code. Can you add an example just to show what it does?